pkg/util/stmtsummary/reader.go

```diff
@@ -64,6 +64,27 @@ func NewStmtSummaryReader(user *auth.UserIdentity, hasProcessPriv bool, cols []*
 	return reader
 }
 
+// GetStmtSummaryCumulativeRows gets statement summary rows with cumulative metrics.
+func (ssr *stmtSummaryReader) GetStmtSummaryCumulativeRows() [][]types.Datum {
```

Comment:
Can we add some input parameters to only needed stats back？like only return records after some ts

Reply:
Hmm, that is a good idea, but I'm not sure if it makes sense in this context. For in-memory tables like this, I think the kind of filtering you're referring to would be done at a higher level, by the executor.

Perhaps we could implement a "pushdown" mechanism (like predicate pushdown to TiKV) for in-memory tables like this. While that could be a useful feature, I think building something like that would be a pretty large project, and is thus out of scope for this PR.